x-pack/plugins/security_solution/server/lib/telemetry/filterlists/prebuilt_rules_alerts.ts

```diff
@@ -253,6 +256,86 @@ export const prebuiltRuleAllowlistFields: AllowlistFields = {
       setting: {
         name: true,
       },
+      application: {
+        name: true,
+      },
+      old_value: true,
+      role: {
+        name: true,
+      },
+    },
+    event: {
+      type: true,
+    },
+  },
+  // kubernetes
+  kubernetes: {
+    audit: {
+      annotations: true,
+      verb: true,
+      user: {
+        username: true,
+        groups: true,
+      },
+      impersonatedUser: {
+        username: true,
+        groups: true,
```

Comment:
Totally get that these are useful for assessing escalation, impersonation, etc. Do we want to hash + salt these to address potential PII concerns?

Reply:
Hey, are we able to filter for all the default accounts beginning with `system:*` and the default user-facing accounts `cluster-admin` , `admin` , `edit` , `view` . Any usernames beyond these we could hash+salt to avoid PII concerns.

Reply:
also `elastic-agent` which is the service account created when deploying the agent

Reply:
I was halfway through implementing a PII scrubbing function and then I realized that we won't be able to release an artifact now or in the future to correspond with this. This would cause us very significant maintenance overhead to maintain artifacts < 8.8 and > 8.8, which opens the doors for all sorts of failure modes.

Could we just keep it simple and drop the `user` and `impersonatedUser`?

cc @elastic/security-data-analytics

Reply:
That's not going to be helpful for rule tuning. Kubernetes rarely uses actual users. Most of them are system and service accounts, it's how things get done in the cluster. We have 3 rules in particular that need these fields as part of the query itself and the others we'd need that field to determine what is normal behavior for tuning purposes.

Reply:
Okok, let me discuss this with my team as we will need to do some additional things on our end to make this work and not leak PII from older releases.

Reply:
Thank you!

Reply:
Had a discussion with @imays11 in our SDA office hours -- the cloud security team is interested in tracking system user behavior to determine if behavior in k8s system namespace is doing malicious/anomalous behavior. This is a special case in that no end-user behavior is expected with these users.

Reply:
Okay, thanks for the follow-up discussion - I will get this implemented and released for 8.7.2 (9 May) and 8.8.0 (21 May)